drivers/soundwire/bus.c

```diff
@@ -500,16 +500,16 @@ int sdw_nread(struct sdw_slave *slave, u32 addr, size_t count, u8 *val)
 {
 	int ret;
 
-	ret = pm_runtime_get_sync(slave->bus->dev);
+	ret = pm_runtime_get_sync(&slave->dev);
 	if (ret < 0 && ret != -EACCES) {
-		pm_runtime_put_noidle(slave->bus->dev);
+		pm_runtime_put_noidle(&slave->dev);
```

Comment:
@plbossart I don't know if it is a right change. As you said in https://github.com/thesofproject/linux/pull/2521, there is a dead lock between regcache_cache_only and regmap_read. The codec driver's logic is that it doesn't expect any IO access in suspend even the register is volatile. And that's why we set cache only in suspend and sync cache in resume. We will get XX when we try to read a register which is unavailable and it is a normal behavior of regmap. Codec driver should resume codec device before regmap_read if it want to read a volatile register in suspend.

Reply:
it's absolutely the right change. We should NOT deal with the parent when playing with the codec. That is a recipe for not waiting for the resume to complete.

I agree there is a conceptual issue with regmap but hiding it by changing the power state at a higher level is completely wrong. We've done this before and it was really bad.

Reply:
Hmm, I don't have the full soundwire context, but I agree this seems like the right thing and is aligned with HDA (e.g. in hdac_device.c we do not power up/down the bus, but rather the device).